lib/components/CircuitJsonPreview/CircuitJsonPreview.tsx

```diff
@@ -89,7 +93,28 @@ export const CircuitJsonPreview = ({
 }: PreviewContentProps) => {
   useStyles()
 
-  const [activeTab, setActiveTabState] = useState(defaultActiveTab ?? "pcb")
+  const [circuitJsonErrors, setCircuitJsonErrors] = useState<
+    CircuitJsonErrors[] | null
+  >(null)
+
+  useEffect(() => {
+    if (circuitJson) {
+      const extractedErrors = circuitJson
+        .filter((e: any) => e.type?.includes("error") && e.error_type)
+        .map((e: any) => ({
+          error: e.error_type,
+          errorMessage: e.message,
+        }))
+
+      setCircuitJsonErrors(extractedErrors.length > 0 ? extractedErrors : null)
+    } else {
+      setCircuitJsonErrors(null)
+    }
+  }, [circuitJson])
```

Comment:
useEffect / useState antipattern. Dont do this!!!!!

just use useMemo. useEffect and useState is _harmful_, it always causes unnecessary rerenders